src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/UpdateAzureMLWebService.cs

```diff
@@ -0,0 +1,195 @@
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New realtime endpoint runtime settings for the web service.")]
+        public RealtimeConfiguration RealtimeConfiguration { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated assets for the web service.")]
+        public IDictionary<string, AssetItem> Assets { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated input schema for the web service.")]
+        public ServiceInputOutputSpecification Input { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated output schema for the web service.")]
+        public ServiceInputOutputSpecification Output { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated global parameter values definition for the web service.")]
+        public IDictionary<string, string> Parameters { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Updated graph package for the web service.")]
+        public GraphPackage Package { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromObjectParameterSet, 
+            Mandatory = true, 
+            HelpMessage = "An updated definition object to update the referenced web service with.", 
+            ValueFromPipeline = true)]
+        public WebService ServiceUpdates { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        protected override void RunCmdlet()
+        {
+            bool isUpdateToReadonly = this.IsReadOnly.IsPresent;
+            if (string.Equals(
+                        this.ParameterSetName, 
+                        UpdateAzureMLWebService.UpdateFromObjectParameterSet, 
+                        StringComparison.OrdinalIgnoreCase))
+            {
+                isUpdateToReadonly = this.ServiceUpdates.Properties != null &&
+                                     this.ServiceUpdates.Properties.ReadOnlyProperty.HasValue &&
+                                     this.ServiceUpdates.Properties.ReadOnlyProperty.Value;
+            }
+
+            if (isUpdateToReadonly)
+            {
+                this.ConfirmAction(
```

Comment:
See previous comment.  You do not need a Force in this case unless you want to prevent overwriting certain properties in some circumstances.  Most likely you justw ant no Force parameter and:

```
if (ShouldProcess(target, action)) { ... actions ...}
```